org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/rewrite/describing/ASTRewritingEitherOrMultiPatternNodeTest.java

```diff
@@ -1569,4 +1573,120 @@ case SampleNotification(String _), PagerNotification1(String _), XNotification _
 		preview= evaluateRewrite(cu, rewrite);
 		assertEqualString(preview, replaceMultipleNodesEndCode);
 	}
+
+	//https://github.com/eclipse-jdt/eclipse.jdt.core/issues/2623
+	public void test0007_a() throws Exception {
+		IPackageFragment pack1= this.sourceFolder.createPackageFragment("test1", false, null);
+		String baseCode = """
+				package x;
+				public class X {
+					public static void main(Object object) {
+						if (object instanceof Path(Pos(int x1, int y1),Pos _)) {
+								//System.out.printf("object is a path starting at x = %d, y = %d%n", x1, y1);
+								System.out.println("abc");
+						}
+					}
+				}
+				record Pos(int x, int y) {}
+				record Path(Pos p1, Pos p2) {}
+				""";
+		ICompilationUnit cu= pack1.createCompilationUnit("X.java", baseCode, false, null);
+		CompilationUnit astRoot= createAST(cu);
+		ASTRewrite rewrite= ASTRewrite.create(astRoot.getAST());
+		AST ast = AST.newAST(AST.JLS22, true);
+
+		assertTrue("Parse errors", (astRoot.getFlags() & ASTNode.MALFORMED) == 0);
+		TypeDeclaration type= findTypeDeclaration(astRoot, "X");
+		MethodDeclaration methodDecl= findMethodDeclaration(type, "main");
+		Block block= methodDecl.getBody();
+		List<?> blockStatements= block.statements();
+		IfStatement ifStatement = (IfStatement) blockStatements.get(0);
+		PatternInstanceofExpression expression = (PatternInstanceofExpression) ifStatement.getExpression();
+		RecordPattern recordPattern = (RecordPattern) expression.getPattern();
+		List<?> patterns = recordPattern.patterns();
+		TypePattern typePattern = (TypePattern) patterns.get(1);
+		{
+			VariableDeclarationFragment vdf = ast.newVariableDeclarationFragment();
+			vdf.setName(ast.newSimpleName("_"));
+			TypePattern newTypePattern = ast.newTypePattern();
+			newTypePattern.setPatternVariable(vdf);
+			rewrite.replace(typePattern, newTypePattern, null);
+		}
+		String ASTConvertedCode = """
+				package x;
+				public class X {
+					public static void main(Object object) {
+						if (object instanceof Path(Pos(int x1, int y1),_)) {
+								//System.out.printf("object is a path starting at x = %d, y = %d%n", x1, y1);
+								System.out.println("abc");
+						}
+					}
+				}
+				record Pos(int x, int y) {}
+				record Path(Pos p1, Pos p2) {}
+				""";
+		String preview= evaluateRewrite(cu, rewrite);
+		assertEqualString(preview, ASTConvertedCode);
+
+	}
+
+	//https://github.com/eclipse-jdt/eclipse.jdt.core/issues/2623
+	public void test0007_b() throws Exception {
+		IPackageFragment pack1= this.sourceFolder.createPackageFragment("test1", false, null);
+		String baseCode = """
+				package x;
+				public class X {
+					public static void main(Object object) {
+						if (object instanceof Path(Pos(int x1, int y1), _)) {
+								//System.out.printf("object is a path starting at x = %d, y = %d%n", x1, y1);
+								System.out.println("abc");
```

Comment:
in the end we don't want any sysouts, not even commented ones :)

Reply:
Done. Removed the commented sysouts

Reply:
sorry, I didn't see that this is not production code, but a test example, so sysouts used for testing purpose is of course good. But I can still see commented print statements that don't seem to serve any purpose.

Reply:
You are right, I have removed all the commented sysouts except one place. Now, I think the code is clean.